### PR TITLE
feat: migrate deprecated classes and custom properties

### PR DESCRIPTION
**Pull request by Blackbaud-ErikaMcVey**
## Summary by CodeRabbit

* **New Features**
  * Added a migration that automatically replaces deprecated CSS variables and class names across HTML, JS/TS, and CSS/SCSS files.
  * Registered the migration in package migration configuration so it can be run as part of updates.

* **Tests**
  * Added comprehensive end-to-end tests validating replacements, boundary/edge cases, mixed updates, ignored file types, and fallback source-root behavior.

* **Chores**
  * Bumped design-tokens dependency and enabled JSON module imports in TypeScript config.

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/4338/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_

Apps:

- [code-examples-playground](https://blackbaud.github.io/skyux-pr-preview/4338/code-examples-playground/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/4338/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/4338/playground/)

File: libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+ */
+function applyCustomPropertyReplacements(
+  content: string,
+  replacements: Record<string, string>,
+): string {
+  let updated = content;
+
+  for (const [oldProp, newProp] of Object.entries(replacements)) {
+    const pattern = new RegExp(escapeRegExp(oldProp), 'g');
+    updated = updated.replace(pattern, newProp);
```
[the remark this answers is not in the document]
Comment (Blackbaud-ErikaMcVey) [reply libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts]: I addressed this, please verify

File: libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+function escapeRegExp(str: string): string {
+  return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
+/**
+ * Replaces CSS class names using word-boundary-aware matching that accounts for
+ * hyphenated class names. Ensures `sky-old-class` does not match inside
+ * `not-sky-old-class` or `sky-old-class-extended`.
+ */
+function applyClassReplacements(
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts]: This function and the `applyCustomPropertyReplacements` function are identical in their implementations. Consider consolidating into a single `applyReplacements` function.

File: libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  content: string,
+  replacements: Record<string, string>,
+): string {
+  let updated = content;
+
+  for (const [oldClass, newClass] of Object.entries(replacements)) {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts]: Every file is run through every regex, even when the file has zero matches. For a workspace with hundreds of files and potentially hundreds of design-token entries, this is unnecessary work. Add a cheap `includes()` guard:
```
if (!updated.includes(oldClass)) {
  continue;
 }
```

File: libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+          updated = applyCustomPropertyReplacements(
+            updated,
+            customPropertyReplacements,
+          );
+
+          if (updated !== content) {
+            const recorder = tree.beginUpdate(filePath);
+            recorder.remove(0, content.length);
+            recorder.insertLeft(0, updated);
+            tree.commitUpdate(recorder);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts]: The recorder is useful for surgical edits, but this is basically a complete file replacement. You're probably safe to replace all four lines with:
```
tree.overwrite(filePath, updated);
```

File: libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    }
+  }
+  for (const group of node.groups ?? []) {
+    traverseTokens(group, result);
+  }
+}
+
+function buildClassReplacements(): Record<string, string> {
+  const result: Record<string, string> = {};
+  traverseClasses(stylesJson as unknown as PublicApiStyles, result);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts]: Is the double assertion `as unknown` needed here? If so, why?

File: libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+  return result;
+}
+
+function buildTokenReplacements(): Record<string, string> {
+  const result: Record<string, string> = {};
+  traverseTokens(tokensJson as unknown as PublicApiTokens, result);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts]: Ditto.

File: libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts (hunk 2 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.ts]: Additionally, since the two functions are identical, I think it would be worth adding a test where a class name and property name have overlap. For example: a file containing both a deprecated class name and a deprecated CSS variable that share a substring (e.g., `sky-color` as a class and `--sky-color` as a property), and write a test that validates the two passes don't interfere.

File: libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.spec.ts [hunk 7]
```diff
@@ -0,0 +1,605 @@
...
+const COLLECTION_PATH = path.join(__dirname, '../../../../../migrations.json');
+const SCHEMATIC_NAME = 'replace-deprecated-css-vars-and-classes';
+
+async function setup(): Promise<{
+  runner: SchematicTestRunner;
+  runSchematic: () => Promise<UnitTestTree>;
+  tree: UnitTestTree;
+}> {
+  const runner = new SchematicTestRunner('migrations', COLLECTION_PATH);
+  const tree = await createTestApp(runner, { projectName: 'my-app' });
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/migrations/update-14/replace-deprecated-css-vars-and-classes/replace-deprecated-css-vars-and-classes.spec.ts:26]: Can you write a test that uses the `createTestLibrary` scaffold to ensure the schematic works on multi-project workspaces?

**Review by Blackbaud-SteveBrush — APPROVED** [review]
